### PR TITLE
Step 8: Jaydel Bujawe Pull Request

### PR DESCRIPTION
**Pull request by jidbujawe**
Hello,

I would like to open a pull request as instructed from Laravel Skills Exercise, Step 8.

Please let me know if I missed anything or did something wrong.

Kind regards,

Jaydel

### CODE REVIEW

File: database/migrations/2021_03_08_031840_create_posts_table.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('posts', function (Blueprint $table) {
+            $table->increments('id');
+            $table->integer('user_id')->unsigned()->index();
+            $table->foreign('user_id')->references('id')->on('users');
```
Comment (stevethomas) [inline database/migrations/2021_03_08_031840_create_posts_table.php:19]: These two line seem to be creating the same column; one or the other would do.

File: app/Http/Controllers/PostController.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Post;
+use Auth;
+
+class PostController extends Controller
+{ 
+    public function createPost(Request $request)
```
Comment (stevethomas) [inline app/Http/Controllers/PostController.php:11]: convention would be call this method 

`public function store(Request $request)`

**Review by stevethomas — APPROVED** [review]
This is a pretty solid first attempt. Well done!

File: app/Http/Controllers/PostController.php (hunk 2 above)
Comment (jidbujawe) [reply app/Http/Controllers/PostController.php:11]: I'll make sure to remember that!

File: database/migrations/2021_03_08_031840_create_posts_table.php (hunk 1 above)
Comment (stevethomas) [reply database/migrations/2021_03_08_031840_create_posts_table.php:19]: To be honest I don't even use foreign key constraints, although the second line probably does what you want. 

You can and should validate the existence of any database IDs within the app where it is coming from user input.

In this instance `request()->user()->id` is validated by Laravel earlier in the request lifecycle so we shouldn't consider it to be user input.

File: app/Http/Controllers/PostController.php (hunk 2 above)
Comment (stevethomas) [reply app/Http/Controllers/PostController.php:11]: See Resource controllers in the Laravel docs; this is what a lot of controllers end up looking like.

Comment (jidbujawe) [reply app/Http/Controllers/PostController.php:11]: I checked out the documentation just now and it made a lot of sense. I'll make sure to follow that convention as well :) Thank you for pointing that out

File: database/migrations/2021_03_08_031840_create_posts_table.php (hunk 1 above)
Comment (jidbujawe) [reply database/migrations/2021_03_08_031840_create_posts_table.php:19]: My apologies I don't understand much of the constraints and laravel validation, I'll make sure to read more about that as well!

**Pull request closed without merging**